fileio/simpleio.go

```diff
@@ -88,7 +89,32 @@ func CatchErrThrowEOF(err error) {
 // Close will return an error if it has already been called.
 func (reader *SimpleReader) Close() {
 	if reader != nil {
-		err := reader.file.Close()
-		common.ExitIfError(err)
+		common.ExitIfError(reader.close())
 	}
 }
+
+// StringToIntSlice will process a column data separated by commas, convert the slice into a slice of type int.
+func StringToIntSlice(column string) []int {
+	work := strings.Split(column, ",")
+	var answer []int = make([]int, len(work))
+	for i := 0; i < len(work)-1; i++ {
+		answer[i] = common.StringToInt(work[i])
+	}
+	return answer
+}
+
+// IntListToString will process a slice of type int as an input and return a each value separated by a comma as a string.
+func IntSliceToString(nums []int) string {
+	ans := strings.Builder{}
```

Comment:
Nice seeing strings.Builder!

Reply:
could someone explain a bit about the strings.Builder usage in code review tomorrow? I haven't seen this before

Reply:
It's basically like a bytes.Buffer but optimized for strings and the most desirable whenever you want to concatenate strings . In a slightly less complicated version, you can think about a string being a slice of bytes `[]byte`. When you want to append data you can to resize the `len([]byte)` and perform memory allocations. String builder optimizes the speed and performance of all this going on under the hood when you build simple strings.